compiler/src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -632,6 +647,11 @@ class Scala2Unpickler(bytes: Array[Byte], classRoot: ClassDenotation, moduleClas
                 // This is the `def` of an accessor that needs to be transformed into
                 // a `val`/`var`. Note that the `Method | Accessor` flags were already
                 // striped away in `readDisambiguatedSymbol`.
+                //
+                // If there was a `<xyz>$access$<idx>` case accessor, we also need to
+                // make this accessor a case accessor.
+                if paramAccessorNeedsCaseAccessor(denot.owner.asClass, denot.name) then
+                  denot.setFlag(CaseAccessor)
```

Comment:
This is setting the flag too late. We have failures with mirrors where we seem to access the symbol flags before we complete the signature.

```scala
    tests/run/typeclass-derivation3.scala failed
    tests/run/i13332intersection.scala failed
```